circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/data/DefaultDataManipulationClientFactoryManagerTest.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (C) 2016-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.core.data;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.hadoop.fs.Path;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.hotels.bdp.circustrain.api.data.DataManipulationClient;
+import com.hotels.bdp.circustrain.api.data.DataManipulationClientFactory;
+import com.hotels.bdp.circustrain.api.data.DataManipulationClientFactoryManager;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DefaultDataManipulationClientFactoryManagerTest {
+
+  private DataManipulationClientFactory s3s3ClientFactory;
+  private DataManipulationClientFactory s3MapReduceClientFactory;
+  private DataManipulationClientFactory hdfsClientFactory;
+
+  private DataManipulationClientFactoryManager manager;
+  private DataManipulationClientFactory clientFactory;
+  private Path sourceLocation;
+  private Path replicaLocation;
+  private final String s3Path = "s3://<path>";
+  private final String hdfsPath = "hdfs://<path>";
+  private final Map<String, Object> copierOptions = new HashMap<>();
+
+  @Before
+  public void setup() {
+    s3s3ClientFactory = new TestDataManipulationClientFactory();
+    ((TestDataManipulationClientFactory) s3s3ClientFactory).setS3Client();
```

Comment:
Might be cleaner to have a small enum with the types S3_S3, S3_MAPREDUCE and HDFS and then have one method called something like `setManipulatorType`. You could also pass the type in to the constructor and then not need this extra method.